File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/humanResources/model/event/PersonCreated.java
package de.vkb.dojo.es.humanResources.model.event;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-created")
public class PersonCreated extends PersonEvent {
    private final String operationId;
    private final String aggregateId;

    private final String username;
    private final String fullname;

    @JsonCreator
    public PersonCreated(
            @JsonProperty("operationId") String operationId,
            @JsonProperty("aggregateId") String aggregateId,
            @JsonProperty("username") String username,
            @JsonProperty("fullname") String fullname
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
        this.username = username;
        this.fullname = fullname;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }

    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }

    @JsonGetter
    public String getUsername() {
        return username;
    }

    @JsonGetter
    public String getFullname() {
        return fullname;
    }
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/model/event/PersonCreated.kt
package de.vkb.dojo.es.humanResources.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-created")
data class PersonCreated(
    override val operationId: String,
    override val aggregateId: String,
    val username: String,
    val fullname: String
): PersonEvent
